### ISSUE TITLE
Build failed at "Built target kokkoskernel"

### ISSUE THREAD

**Issue by lkkevin** [trilinos/Trilinos#1998]
Hi, 

I am trying to build Trilinos 12.12.1 for Peridigm on CentOS 7.4.  The "make -j 2" failed at [59] Built target kokkoskernel.  I have been working on this issue for a couple day but could not figure out.   Please help.  Thanks.

#------ These are the prerequisites that i installed on my VM: -----#
boost_1_65_1 ,  cmake-3.10.0-rc5,  hdf5-1.10.1,  matio-1.5.11,  netcdf-c-4.5.0,  openmpi-3.0.0,  blas 3.8.0,  lapack 3.8.0.

#------ This is the CMakeError.log ------#
Performing C++ SOURCE FILE Test HAVE_TEUCHOS_LAPACKLARND failed with the following output:
Change Dir: /root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/gmake" "cmTC_2fa60/fast"
/usr/bin/gmake -f CMakeFiles/cmTC_2fa60.dir/build.make CMakeFiles/cmTC_2fa60.dir/build
gmake[1]: Entering directory `/root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp'
Building CXX object CMakeFiles/cmTC_2fa60.dir/src.cxx.o
/usr/local/bin/mpicxx    -O2 -std=c++11 -pedantic -ftrapv -Wall -Wno-long-long -std=c++11 -DHAVE_TEUCHOS_LAPACKLARND -O3 -DNDEBUG   -o CMakeFiles/cmTC_2fa60.dir/src.cxx.o -c /root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp/src.cxx
/root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp/src.cxx: In function ‘int main()’:
/root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp/src.cxx:12:38: warning: narrowing conversion of ‘0.0’ from ‘double’ to ‘int’ inside { } [-Wnarrowing]
   int seed[4] = { 0.0, 0.0, 0.0, 1.0 };
                                      ^
/root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp/src.cxx:12:38: warning: narrowing conversion of ‘0.0’ from ‘double’ to ‘int’ inside { } [-Wnarrowing]
/root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp/src.cxx:12:38: warning: narrowing conversion of ‘0.0’ from ‘double’ to ‘int’ inside { } [-Wnarrowing]
/root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp/src.cxx:12:38: warning: narrowing conversion of ‘1.0e+0’ from ‘double’ to ‘int’ inside { } [-Wnarrowing]
Linking CXX executable cmTC_2fa60
/usr/local/bin/cmake -E cmake_link_script CMakeFiles/cmTC_2fa60.dir/link.txt --verbose=1
/usr/local/bin/mpicxx    -O2 -std=c++11 -pedantic -ftrapv -Wall -Wno-long-long -std=c++11 -DHAVE_TEUCHOS_LAPACKLARND -O3 -DNDEBUG    -rdynamic CMakeFiles/cmTC_2fa60.dir/src.cxx.o  -o cmTC_2fa60 /usr/local/lib/liblapack.a
CMakeFiles/cmTC_2fa60.dir/src.cxx.o: In function `main':
src.cxx:(.text.startup+0x22): undefined reference to `dlarnd_'
collect2: error: ld returned 1 exit status
gmake[1]: *** [cmTC_2fa60] Error 1
gmake[1]: Leaving directory `/root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp'
gmake: *** [cmTC_2fa60/fast] Error 2

Return value: 1
Source file was:

#define F77_BLAS_MANGLE(name,NAME) name ## _
#define DLARND_F77   F77_BLAS_MANGLE(dlarnd,DLARND)
extern "C" { double DLARND_F77(const int* idist, int* seed); }
int main()
{

  const int idist = 1;
  int seed[4] = { 0.0, 0.0, 0.0, 1.0 };
  double val = DLARND_F77(&idist, seed);
  return (val < 0.0 ? 1 : 0);
}
Performing C++ SOURCE FILE Test HAVE_CXX_PRAGMA_WEAK failed with the following output:
Change Dir: /root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/gmake" "cmTC_82e34/fast"
/usr/bin/gmake -f CMakeFiles/cmTC_82e34.dir/build.make CMakeFiles/cmTC_82e34.dir/build
gmake[1]: Entering directory `/root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp'
Building CXX object CMakeFiles/cmTC_82e34.dir/src.cxx.o
/usr/local/bin/mpicxx    -O2 -std=c++11 -pedantic -ftrapv -Wall -Wno-long-long -std=c++11 -DHAVE_CXX_PRAGMA_WEAK -O3 -DNDEBUG   -o CMakeFiles/cmTC_82e34.dir/src.cxx.o -c /root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp/src.cxx
Linking CXX executable cmTC_82e34
/usr/local/bin/cmake -E cmake_link_script CMakeFiles/cmTC_82e34.dir/link.txt --verbose=1
/usr/local/bin/mpicxx    -O2 -std=c++11 -pedantic -ftrapv -Wall -Wno-long-long -std=c++11 -DHAVE_CXX_PRAGMA_WEAK -O3 -DNDEBUG    -rdynamic CMakeFiles/cmTC_82e34.dir/src.cxx.o  -o cmTC_82e34
CMakeFiles/cmTC_82e34.dir/src.cxx.o: In function `main':
src.cxx:(.text.startup+0x1c): undefined reference to `A::theFunction()'
collect2: error: ld returned 1 exit status
gmake[1]: *** [cmTC_82e34] Error 1
gmake[1]: Leaving directory `/root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp'
gmake: *** [cmTC_82e34/fast] Error 2

Source file was:

#include <iostream>

namespace A {
// theFunction never gets defined, because we
// don't link with a library that defines it.
// That's OK, because it's weak linkage.
#pragma weak theFunction
extern void theFunction ();
}

int main() {
  std::cout << "Hi!  I am main." << std::endl;
  if (A::theFunction != NULL) {
    // Should never be called, since we don't link
    // with a library that defines A::theFunction.
    A::theFunction ();
  }
  return 0;
}

Determining if the function dggsvd3 exists failed with the following output:
Change Dir: /root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp
Run Build Command:"/usr/bin/gmake" "cmTC_427b3/fast"
/usr/bin/gmake -f CMakeFiles/cmTC_427b3.dir/build.make CMakeFiles/cmTC_427b3.dir/build
gmake[1]: Entering directory `/root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_427b3.dir/CheckFunctionExists.c.o
/usr/local/bin/mpicc   -DCHECK_FUNCTION_EXISTS=dggsvd3 -O3 -DNDEBUG   -o CMakeFiles/cmTC_427b3.dir/CheckFunctionExists.c.o   -c /usr/local/share/cmake-3.10/Modules/CheckFunctionExists.c
Linking C executable cmTC_427b3
/usr/local/bin/cmake -E cmake_link_script CMakeFiles/cmTC_427b3.dir/link.txt --verbose=1
/usr/local/bin/mpicc  -DCHECK_FUNCTION_EXISTS=dggsvd3 -O3 -DNDEBUG    -rdynamic CMakeFiles/cmTC_427b3.dir/CheckFunctionExists.c.o  -o cmTC_427b3 /usr/local/lib/liblapack.a /usr/local/lib/libblas.a
CMakeFiles/cmTC_427b3.dir/CheckFunctionExists.c.o: In function `main':
CheckFunctionExists.c:(.text.startup+0xc): undefined reference to `dggsvd3'
collect2: error: ld returned 1 exit status
gmake[1]: *** [cmTC_427b3] Error 1
gmake[1]: Leaving directory `/root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp'
gmake: *** [cmTC_427b3/fast] Error 2

Determining if the function dggsvd3_ exists failed with the following output:
Change Dir: /root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/gmake" "cmTC_01e16/fast"
/usr/bin/gmake -f CMakeFiles/cmTC_01e16.dir/build.make CMakeFiles/cmTC_01e16.dir/build
gmake[1]: Entering directory `/root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_01e16.dir/CheckFunctionExists.c.o
/usr/local/bin/mpicc   -DCHECK_FUNCTION_EXISTS=dggsvd3_ -O3 -DNDEBUG   -o CMakeFiles/cmTC_01e16.dir/CheckFunctionExists.c.o   -c /usr/local/share/cmake-3.10/Modules/CheckFunctionExists.c
Linking C executable cmTC_01e16
/usr/local/bin/cmake -E cmake_link_script CMakeFiles/cmTC_01e16.dir/link.txt --verbose=1
/usr/local/bin/mpicc  -DCHECK_FUNCTION_EXISTS=dggsvd3_ -O3 -DNDEBUG    -rdynamic CMakeFiles/cmTC_01e16.dir/CheckFunctionExists.c.o  -o cmTC_01e16 /usr/local/lib/liblapack.a /usr/local/lib/libblas.a
/usr/bin/ld: /usr/local/lib/liblapack.a(dlaqps.o): undefined reference to symbol 'lround@@GLIBC_2.2.5'
/usr/lib64/libm.so.6: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
gmake[1]: *** [cmTC_01e16] Error 1
gmake[1]: Leaving directory `/root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp'
gmake: *** [cmTC_01e16/fast] Error 2

Determining if the function DGGSVD3 exists failed with the following output:
Change Dir: /root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp
Run Build Command:"/usr/bin/gmake" "cmTC_e1f49/fast"
/usr/bin/gmake -f CMakeFiles/cmTC_e1f49.dir/build.make CMakeFiles/cmTC_e1f49.dir/build
gmake[1]: Entering directory `/root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_e1f49.dir/CheckFunctionExists.c.o
/usr/local/bin/mpicc   -DCHECK_FUNCTION_EXISTS=DGGSVD3 -O3 -DNDEBUG   -o CMakeFiles/cmTC_e1f49.dir/CheckFunctionExists.c.o   -c /usr/local/share/cmake-3.10/Modules/CheckFunctionExists.c
Linking C executable cmTC_e1f49
/usr/local/bin/cmake -E cmake_link_script CMakeFiles/cmTC_e1f49.dir/link.txt --verbose=1
/usr/local/bin/mpicc  -DCHECK_FUNCTION_EXISTS=DGGSVD3 -O3 -DNDEBUG    -rdynamic CMakeFiles/cmTC_e1f49.dir/CheckFunctionExists.c.o  -o cmTC_e1f49 /usr/local/lib/liblapack.a /usr/local/lib/libblas.a
CMakeFiles/cmTC_e1f49.dir/CheckFunctionExists.c.o: In function `main':
CheckFunctionExists.c:(.text.startup+0xc): undefined reference to `DGGSVD3'
collect2: error: ld returned 1 exit status
gmake[1]: *** [cmTC_e1f49] Error 1
gmake[1]: Leaving directory `/root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp'
gmake: *** [cmTC_e1f49/fast] Error 2

Determining if the function DGGSVD3_ exists failed with the following output:
Change Dir: /root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/gmake" "cmTC_6d0e5/fast"
/usr/bin/gmake -f CMakeFiles/cmTC_6d0e5.dir/build.make CMakeFiles/cmTC_6d0e5.dir/build
gmake[1]: Entering directory `/root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_6d0e5.dir/CheckFunctionExists.c.o
/usr/local/bin/mpicc   -DCHECK_FUNCTION_EXISTS=DGGSVD3_ -O3 -DNDEBUG   -o CMakeFiles/cmTC_6d0e5.dir/CheckFunctionExists.c.o   -c /usr/local/share/cmake-3.10/Modules/CheckFunctionExists.c
Linking C executable cmTC_6d0e5
/usr/local/bin/cmake -E cmake_link_script CMakeFiles/cmTC_6d0e5.dir/link.txt --verbose=1
/usr/local/bin/mpicc  -DCHECK_FUNCTION_EXISTS=DGGSVD3_ -O3 -DNDEBUG    -rdynamic CMakeFiles/cmTC_6d0e5.dir/CheckFunctionExists.c.o  -o cmTC_6d0e5 /usr/local/lib/liblapack.a /usr/local/lib/libblas.a
CMakeFiles/cmTC_6d0e5.dir/CheckFunctionExists.c.o: In function `main':
CheckFunctionExists.c:(.text.startup+0xc): undefined reference to `DGGSVD3_'
collect2: error: ld returned 1 exit status
gmake[1]: *** [cmTC_6d0e5] Error 1
gmake[1]: Leaving directory `/root/Downloads/trilinos-12.12.1-Source/build/CMakeFiles/CMakeTmp'
gmake: *** [cmTC_6d0e5/fast] Error 2

#------ This is my build.sh -----#

#!/bin/bash
#===========================================================
make clean
rm -rf CMake*

cmake \
 -D CMAKE_INSTALL_PREFIX:PATH=/usr/local/trilinos \
 -D MPI_BASE_DIR:PATH=/usr/local/lib/openmpi \
 -D CMAKE_CXX_FLAGS:STRING="-O2 -std=c++11 -pedantic -ftrapv -Wall -Wno-long-long" \
 -D CMAKE_BUILD_TYPE:STRING=RELEASE \
 -D Trilinos_WARNINGS_AS_ERRORS_FLAGS:STRING="" \
 -D Trilinos_ENABLE_ALL_PACKAGES:BOOL=OFF \
 -D Trilinos_ENABLE_Teuchos:BOOL=ON \
 -D Trilinos_ENABLE_Shards:BOOL=ON \
 -D Trilinos_ENABLE_Sacado:BOOL=ON \
 -D Trilinos_ENABLE_Epetra:BOOL=ON \
 -D Trilinos_ENABLE_EpetraExt:BOOL=ON \
 -D Trilinos_ENABLE_Ifpack:BOOL=ON \
 -D Trilinos_ENABLE_AztecOO:BOOL=ON \
 -D Trilinos_ENABLE_Amesos:BOOL=ON \
 -D Trilinos_ENABLE_Anasazi:BOOL=ON \
 -D Trilinos_ENABLE_Belos:BOOL=ON \
 -D Trilinos_ENABLE_ML:BOOL=ON \
 -D Trilinos_ENABLE_Phalanx:BOOL=ON \
 -D Trilinos_ENABLE_Intrepid:BOOL=ON \
 -D Trilinos_ENABLE_NOX:BOOL=ON \
 -D Trilinos_ENABLE_Stratimikos:BOOL=ON \
 -D Trilinos_ENABLE_Thyra:BOOL=ON \
 -D Trilinos_ENABLE_Rythmos:BOOL=ON \
 -D Trilinos_ENABLE_MOOCHO:BOOL=ON \
 -D Trilinos_ENABLE_TriKota:BOOL=OFF \
 -D Trilinos_ENABLE_Stokhos:BOOL=ON \
 -D Trilinos_ENABLE_Zoltan:BOOL=ON \
 -D Trilinos_ENABLE_Piro:BOOL=ON \
 -D Trilinos_ENABLE_Teko:BOOL=ON \
 -D Trilinos_ENABLE_SEACASIoss:BOOL=ON \
 -D Trilinos_ENABLE_SEACAS:BOOL=ON \
 -D Trilinos_ENABLE_SEACASBlot:BOOL=ON \
 -D Trilinos_ENABLE_Pamgen:BOOL=ON \
 -D Trilinos_ENABLE_EXAMPLES:BOOL=OFF \
 -D Trilinos_ENABLE_TESTS:BOOL=ON \
 -D TPL_ENABLE_HDF5:BOOL=ON \
 -D HDF5_INCLUDE_DIRS:PATH=/usr/local/HDF5/include \
 -D HDF5_LIBRARY_DIRS:PATH=/usr/local/HDF5/lib \
 -D TPL_ENABLE_Netcdf:BOOL=ON \
 -D Netcdf_INCLUDE_DIRS:PATH=/usr/local/netcdf/include \
 -D Netcdf_LIBRARY_DIRS:PATH=/usr/local/netcdf/lib \
 -D TPL_ENABLE_MPI:BOOL=ON \
 -D TPL_ENABLE_BLAS:BOOL=ON \
 -D TPL_ENABLE_LAPACK:BOOL=ON \
 -D TPL_ENABLE_Boost:BOOL=ON \
 -D Boost_INCLUDE_DIRS:PATH=/usr/local/boost/include \
 -D Boost_LIBRARY_DIRS:PATH=/usr/local/boost/lib \
 -D BLAS_LIBRARY_NAMES:STRING=libblas.a \
 -D BLAS_LIBRARY_DIRS:STRING=/usr/local/lib \
 -D LAPACK_LIBRARY_NAMES:STRING=liblapack.a \
 -D LAPACK_LIBRARY_DIRS:STRING=/usr/local/lib \
 -D CMAKE_VERBOSE_MAKEFILE:BOOL=OFF \
 -D Trilinos_VERBOSE_CONFIGURE:BOOL=OFF \
 ../
#---------------------------------------------------------------------#

@trilinos/<teamName>

## Your Environment

- **Relevant repo SHA1s:**  
- **Relevant configure flags or configure script:**  
- **Operating system and version:**  
- **Compiler and TPL versions:**  

## Related Issues

* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to 
* Part of 
* Composed of

**Comment by jhux2** [comment]
@trilinos/kokkos-kernels

**Comment by lkkevin** [comment]
Thanks Jhux2.  That links to a page not found.

**Comment by mndevec** [comment]
Could it be related to : https://github.com/trilinos/Trilinos/issues/1818 ?

**Comment by lkkevin** [comment]
Thanks mndevec.   KokkosKernels is a library to match Trilinos style, i believe so.  I dont  know how to resove this error.

**Comment by jhux2** [comment]
@lkkevin Is this happening during the cmake **configure** process?   It looks to me like a cmake error indicating the blas and lapack libraries can't be found.

This looks fishy in your configure:
``` 
-D BLAS_LIBRARY_NAMES:STRING=libblas.a
-D LAPACK_LIBRARY_NAMES:STRING=liblapack.a 
```



> > @trilinos/kokkos-kernels 

> Thanks Jhux2. That links to a page not found.

That's not a page, but a Trilinos team: https://github.com/orgs/trilinos/teams.

**Comment by lkkevin** [comment]
@jhux2 - **configure** went through successful but with 2 failed performance tests:
-- Performing Test HAVE_TEUCHOS_LAPACKLARND - Failed
-- Performing Test HAVE_CXX_PRAGMA_WEAK - Failed

System found blas & lapack library.  I received the "...kokkoskernel" error when running "**make -j 2**"
Thanks for your suggestion.  I can edit the blas & lapack lib path and rerun the build to see if it fixes the error.

**Comment by lkkevin** [comment]
@jhux2 - you are right.  i think BLAS or LAPACK maybe a root cause.  I re-build BLAS and LAPACK, re-run "make -j2", the build went pass the 59% and got stuck at 99% with this error.   Any help is appreciate.  Thanks.
.....
Scanning dependencies of target Stokhos_SDMUtilsUnitTest
[ 99%] Building CXX object packages/stokhos/test/UnitTest/CMakeFiles/Stokhos_SDMUtilsUnitTest.dir/Stokhos_SDMUtilsUnitTest.cpp.o
[ 99%] Linking CXX executable Stokhos_SDMUtilsUnitTest.exe
CMakeFiles/Stokhos_SDMUtilsUnitTest.dir/Stokhos_SDMUtilsUnitTest.cpp.o: In function `void Stokhos::CPQR_Householder<int, double>(Teuchos::SerialDenseMatrix<int, double> const&, Teuchos::SerialDenseMatrix<int, double>&, Teuchos::SerialDenseMatrix<int, double>&, Teuchos::Array<int>&)':
Stokhos_SDMUtilsUnitTest.cpp:(.text._ZN7Stokhos16CPQR_HouseholderIidEEvRKN7Teuchos17SerialDenseMatrixIT_T0_EERS5_S8_RNS1_5ArrayIS3_EE[_ZN7Stokhos16CPQR_HouseholderIidEEvRKN7Teuchos17SerialDenseMatrixIT_T0_EERS5_S8_RNS1_5ArrayIS3_EE]+0x16c): undefined reference to `dgeqpf_'
collect2: error: ld returned 1 exit status
make[2]: *** [packages/stokhos/test/UnitTest/Stokhos_SDMUtilsUnitTest.exe] Error 1
make[1]: *** [packages/stokhos/test/UnitTest/CMakeFiles/Stokhos_SDMUtilsUnitTest.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
In file included from /root/Downloads/trilinos-12.12.1-Source/build/packages/muelu/src/MueLu_VariableDofLaplacianFactory.hpp:2:0,
                 from /root/Downloads/trilinos-12.12.1-Source/packages/muelu/src/Interface/MueLu_FactoryFactory_decl.hpp:146,
                 from /root/Downloads/trilinos-12.12.1-Source/build/packages/muelu/src/Interface/MueLu_FactoryFactory.hpp:1,
                 from /root/Downloads/trilinos-12.12.1-Source/packages/muelu/src/Interface/MueLu_ParameterListInterpreter_def.hpp:73,
                 from /root/Downloads/trilinos-12.12.1-Source/build/packages/muelu/src/Interface/MueLu_ParameterListInterpreter.hpp:2,
                 from /root/Downloads/trilinos-12.12.1-Source/packages/muelu/adapters/tpetra/MueLu_CreateTpetraPreconditioner.hpp:22,
                 from /root/Downloads/trilinos-12.12.1-Source/packages/stokhos/test/UnitTest/Stokhos_TpetraCrsMatrixMPVectorUnitTest.hpp:77,
                 from /root/Downloads/trilinos-12.12.1-Source/packages/stokhos/test/UnitTest/Stokhos_TpetraCrsMatrixMPVectorUnitTest_Serial.cpp:46:
/root/Downloads/trilinos-12.12.1-Source/packages/muelu/src/Interface/../Graph/MueLu_VariableDofLaplacianFactory_def.hpp: In member function ‘void MueLu::VariableDofLaplacianFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::Build(MueLu::VariableDofLaplacianFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::Level&) const [with Scalar = Sacado::MP::Vector<Stokhos::StaticFixedStorage<int, double, 16, Kokkos::Serial> >; LocalOrdinal = int; GlobalOrdinal = int; Node = Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Serial>; MueLu::VariableDofLaplacianFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::Level = MueLu::Level]’:
/root/Downloads/trilinos-12.12.1-Source/packages/muelu/src/Interface/../Graph/MueLu_VariableDofLaplacianFactory_def.hpp:256:16: warning: ‘blockRow’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     amalgRowPtr[blockRow+1] = newNzs;
                ^
[ 99%] Linking CXX executable Stokhos_TpetraCrsMatrixMPVectorUnitTest_Serial.exe
[ 99%] Built target Stokhos_TpetraCrsMatrixMPVectorUnitTest_Serial
make: *** [all] Error 2
You have mail in /var/spool/mail/root
[root@peri build]#

**Comment by lkkevin** [comment]
Hi all, thanks for your input.  I would like to close this issue and open a new thread for a new issue #2005 .  Thanks.

**Comment by jhux2** [comment]
Closing per OP.

**Comment by mhoemmen** [comment]
@lkkevin wrote:
> configure went through successful but with 2 failed performance tests:
> -- Performing Test HAVE_TEUCHOS_LAPACKLARND - Failed
> -- Performing Test HAVE_CXX_PRAGMA_WEAK - Failed

FYI those are not "performance tests."  The first is an optional LAPACK feature.  Some LAPACK libraries build with LAPACK testing routines like `xLARND` (replace x with D, S, etc.); some don't.  The second is an optional compiler feature.  Trilinos does not require compilers to implement `#pragma weak`.

**Comment by lkkevin** [comment]
@mhoemmen - First time building Trilinos, so i did not know.  Thanks for letting me know.

**Comment by mhoemmen** [comment]
@lkkevin No problem!  I really wish CMake would let me print something other than "Failed" there (the xLARND test was my doing, circa 2010).

**Comment by lkkevin** [comment]
@mhoemmen - Again thank you all for your and other contributions.  I and many people really appreciated for your works.

**Comment by mhoemmen** [comment]
@lkkevin Glad to help! :-) We would appreciate feedback from how Peridigm benefits from using the latest Trilinos version.
